src/iterator.cr

```diff
@@ -297,30 +297,26 @@ module Iterator(T)
   # * If *reuse* is given, the array can be reused
   # * If *reuse* is an `Array`, this array will be reused
```

Comment:
This should be updated to allow `Array`, `Deque` or any collection type with a matching interface.

Reply:
In practice, the only ones I found in stdlib that have `<<`, `shift` and `size` seems to be `Array` and `Deque`. I suppose this is where java or whatever would have reached for a Qeueable interface.

I'll synch the texts of the two methods, though.

Reply:
Yeah, the stdlib has only those two matching collections. But it should also be usable with any custom collection type.